File: server/datasets/python_source-code/Multiple-A2/reformatted1_bubble_sort.py
"""
Bubble Sort - reformatted1
Approach: Identical logic and identifier names to unique1 - only
whitespace, indentation, and statement grouping have changed.
"""
from typing import List
class BubbleSorter:
    def __init__(self, data: List[int]) -> None:
        self.data = list(data); self.swaps = 0; self.passes = 0
    def sort(self) -> List[int]:
        n = len(self.data)
        for i in range(n - 1):
            self.passes += 1
            swapped_this_pass = False
            for j in range(n - 1 - i):
                if self.data[j] > self.data[j + 1]:
                    self.data[j], self.data[j + 1] = self.data[j + 1], self.data[j]; self.swaps += 1; swapped_this_pass = True
            if not swapped_this_pass: break
        return self.data
def main() -> None:
    arr = [64, 34, 25, 12, 22, 11, 90, 5, 77, 1]
    sorter = BubbleSorter(arr)
    result = sorter.sort()
    print(f"Sorted: {result}")
    print(f"Swaps: {sorter.swaps}, Passes: {sorter.passes}")
if __name__ == "__main__":
    main()

File: server/datasets/java_source-code/Multiple-A2/reformatted1_bubble_sort.java
/*
 * Bubble Sort - reformatted1
 * Approach: Identical logic and identifier names to unique1 - only
 * whitespace, indentation, and statement grouping have changed.
 */
import java.util.Arrays;
class BubbleSorter {
    int[] data; int swaps; int passes;
    public BubbleSorter(int[] data) { this.data = Arrays.copyOf(data, data.length); this.swaps = 0; this.passes = 0; }
    public int[] sort() {
        int n = this.data.length;
        for (int i = 0; i < n - 1; i++) {
            this.passes++; boolean swappedThisPass = false;
            for (int j = 0; j < n - 1 - i; j++) {
                if (this.data[j] > this.data[j + 1]) {
                    int temp = this.data[j]; this.data[j] = this.data[j + 1]; this.data[j + 1] = temp;
                    this.swaps++; swappedThisPass = true;
                }
            }
            if (!swappedThisPass) { break; }
        }
        return this.data;
    }
    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90, 5, 77, 1};
        BubbleSorter sorter = new BubbleSorter(arr);
        int[] result = sorter.sort();
        System.out.println("Sorted: " + Arrays.toString(result));
        System.out.println("Swaps: " + sorter.swaps + ", Passes: " + sorter.passes);
    }
}
